packaging/sources/preupgrade-assistant-xml-logs.patch

```diff
@@ -0,0 +1,38 @@
+diff --git a/preupg/report_parser.py b/preupg/report_parser.py
+index 1a9047c..7223bf4 100644
+--- a/preupg/report_parser.py
++++ b/preupg/report_parser.py
+@@ -198,6 +198,25 @@ class ReportParser(object):
+ 
+         self.write_xml()
+ 
++    def fix_stderr(self):
++        """The function transform stdout to stderr."""
++        for rule in self.get_all_result_rules():
++        # Filter all check-import=stdout which are empty and remove them
++            for check in self.get_nodes(rule, "check"):
++                cim_stdout = [x for x in self.get_nodes(check, "check-import")
++                                if x.get('import-name') == "stdout"]
++                cim_stderr = [x for x in self.get_nodes(check, "check-import")
++                                if x.get('import-name') == "stderr"]
++                if not cim_stdout or cim_stdout[0].text is None or cim_stdout[0].text == "":
++                    continue
++
++                # unexpected, but just to be sure..
++                if cim_stderr:
++                    if cim_stderr[0].text is not None and cim_stderr[0].text != "":
++                        cim_stdout[0].text +="\n"+cim_stderr[0].text
++                    remove_node(check, cim_stderr[0])
```

Comment:
Removing the stderr node here assumes there's just one stderr node - which will probably be always the case. However the condition is "_if cim_stderr:_" and _cim_stderr_ is a list, so the logic is strange - if there's multiple stderr nodes, just the first one would be removed.
Also I don't like much removing empty nodes. There may be utilities which expect the nodes to be there (for example preupg-dif :). I know, it'd be a fault of the utility to not be robust enough, if it collapses when the expected node's not there, but anyway.
However, keep the code as is due to what I wrote in the comment above.